src/partition/repart.c

```diff
@@ -1316,6 +1320,41 @@ static int config_parse_size4096(
         return 0;
 }
 
+static int config_parse_block_size(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        uint64_t *blksz = data;
+        int r;
+
+        assert(rvalue);
+        assert(data);
+
+        r = safe_atou64(rvalue, blksz);
+        if (r < 0)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                        "Failed to parse size value: %s", rvalue);
+
+        if (*blksz < 512 || *blksz > 4096)
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                        "Value not between 512 and 4096: %s", rvalue);
+
+        if (!ISPOWEROF2(*blksz)) {
+                return log_syntax(unit, LOG_ERR, filename, line, r,
+                        "Value not a power of 2: %s", rvalue);
+        }
```

Comment:
no {} around single-line if blocks, according to our coding style, please

Reply:
Fixed.